pinot-common/src/main/java/com/linkedin/pinot/common/utils/EqualityUtils.java

```diff
@@ -39,12 +40,18 @@ public static boolean isEqual(long left, long right) {
     return left == right;
   }
 
+  /**
+   * This method is not the same as '==' operator, isEqual(Float.NaN, Float.NaN) returns true instead of false.
```

Comment:
This could also be written as "Compares both arguments for equality, and considers NaN to be equal to NaN (unlike the Java == operator)."